memgpt/schemas/block.py

```diff
@@ -0,0 +1,51 @@
+from typing import List, Optional, Union
+
+from pydantic import BaseModel, Field, validator
+
+# block of the LLM context
+
+
+class Block(BaseModel):
+    """Block of the LLM context"""
+
+    description: Optional[str] = Field(None, description="Description of the block.")
+    limit: int = Field(2000, description="Character limit of the block.")
+    value: Optional[Union[List[str], str]] = Field(None, description="Value of the block.")
+
+    def __setattr__(self, name, value):
+        """Run validation if self.value is updated"""
+        super().__setattr__(name, value)
+        if name == "value":
+            # run validation
+            self.__class__.validate(self.dict(exclude_unset=True))
+
+    @validator("value", always=True)
+    def check_value_length(cls, v, values):
+        if v is not None:
+            # Fetching the limit from the values dictionary
+            limit = values.get("limit", 2000)  # Default to 2000 if limit is not yet set
+
+            # Check if the value exceeds the limit
+            if isinstance(v, str):
+                length = len(v)
+            elif isinstance(v, list):
+                length = sum(len(item) for item in v)
+            else:
+                raise ValueError("Value must be either a string or a list of strings.")
+
```

Comment:
I'd probably do something like 
```python
try:
    assert length >= limit, f"Edit failed: Exceeds {limit} character limit (requested {length})."
except AssertionError as e:
    # this is not my favorite pydantic trait; 
    # pydantic validators require a ValueError to be raised to be included as a validation error. They really should handle AssertionErrors IMO, but we do this to make them happy. 
    raise ValueError(e)
return v 
```
That way a value that is too long is included in the Pydantic validation errors instead of a different error you need to format and deal with.